corehq/apps/locations/models.py

```diff
@@ -282,7 +282,7 @@ def couch_locations(self, wrapped=True):
 
     def accessible_to_user(self, domain, user):
         if user.has_permission(domain, 'access_all_locations'):
-            return self.all()
+            return self.filter(domain=domain)
```

Comment:
Agreed! `get_location_ids` shadows unwanted locations. Just that it would make things uniform...
for all access locations in domain and for restricted access the same set & assigned locations